prow/cmd/pipeline/controller.go

```diff
@@ -455,6 +469,21 @@ func reconcile(c reconciler, key string) error {
 			// the pipeline cannot be executed (e.g. referenced pipeline does not exist)
 			return updateProwJobState(c, key, newPipelineRun, pj, newpj, prowjobv1.ErrorState, jerr.Error())
 		}
+
+		// Set pipeline resources owner so that they are deleted when the pipeline run goes away
+		for _, pr := range pipelineResources {
```

Comment:
What about instead just iterating over the `resources`, doing a `Get`, checking if the `OwnerRef` is already there and adding it & `Update` if not?

Reply:
The pipeline resources were created by us a few lines above without an owner.
Why should we make an additional `Get` for something we already have, and check `OwnerRef` when we created the resource without one ?
Do you think about something that could break in this case ?

Reply:
Because we tolerate `IsConflict` errors in which case the pipeline resource might be nil

Reply:
Thinking about it again, i'm going to do it completely differently.

I am not sure when it was added in tekton but we now have the ability to define resources _inline_ in the pipeline run, that would eliminate the need to create pipeline resources and pipeline runs separately.